src/components/Signup/SignupForm.tsx

```diff
@@ -0,0 +1,108 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { AuthError, signUp } from "aws-amplify/auth";
+import Icon from "components/Icon";
+import { Form } from "components/form";
+import { GENERIC_ERROR_MESSAGE } from "constants/common";
+import { useErrorContext } from "contexts/ErrorContext";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { requiredString } from "schemas/string";
+import { object } from "yup";
+import { Donor, StateSetter } from "./types";
+
+type Props = {
+  donor: Donor;
+  setSignupState: StateSetter;
+  classes?: string;
+};
+
+export default function SignupForm(props: Props) {
+  const { handleError } = useErrorContext();
+  const [isPasswordShown, setIsPasswordShown] = useState(false);
+  const {
+    register,
+    formState: { errors, isSubmitting },
+    handleSubmit,
+  } = useForm({
+    resolver: yupResolver(
+      object({
+        password: requiredString
+          .min(8, ({ min }) => `must have at least ${min} characters`)
+          .matches(/[a-z]/, "must have lowercase letters")
+          .matches(/[A-Z]/, "must have uppercase letters")
+          .matches(/\d/, "must have numbers")
+          .matches(
+            /[!@#$%^&*()_+{}\[\]:;<>,.?~\\/-]/,
+            "must have special characters"
+          ),
+      })
+    ),
+  });
+
+  return (
+    <Form
+      className={`${props.classes ?? ""} grid`}
+      disabled={isSubmitting}
+      onSubmit={handleSubmit(async (fv) => {
+        try {
+          const { nextStep } = await signUp({
+            username: props.donor.email,
+            password: fv.password,
+            options: {
+              userAttributes: {
+                given_name: props.donor.firstName,
+                family_name: props.donor.lastName,
+              },
+              autoSignIn: false,
+            },
+          });
+
+          //per cognito config
+          if (nextStep.signUpStep !== "CONFIRM_SIGN_UP") throw "";
```

Comment:
Shouldn't we throw some kind of error message, at least for debugging purposes? We don't have to show it to the users (the `catch` block is already set up correctly)